src/visualization.rs

```diff
@@ -0,0 +1,139 @@
+                .flatten()
+                .collect();
+
+            let line_string_features: Vec<_> = related_stop_ids
+                .iter()
+                .map(|related_stop| {
+                    geojson_feature_line_string(&stop_id, related_stop, gtfs, issue)
+                })
+                .flatten()
+                .collect();
+
+            let features = stop_features
+                .into_iter()
+                .chain(line_string_features.into_iter())
+                .collect();
+
+            let feature_collection = FeatureCollection {
+                bbox: None,
+                features: features,
+                foreign_members: None,
+            };
+
+            Some(feature_collection)
+        }
+        _ => None,
+    }
+}
+
+fn geojson_feature_point(stop_id: &String, gtfs: &Gtfs) -> Option<Feature> {
+    gtfs.stops.get(stop_id).map(|stop| {
+        let stop_geom = get_stop_geom(stop);
+        let mut properties = Map::new();
+
+        properties.insert(String::from("id"), to_value(&stop.id).unwrap());
+        properties.insert(String::from("name"), to_value(&stop.name).unwrap());
+
+        Feature {
+            geometry: stop_geom,
+            bbox: None,
+            properties: Some(properties),
+            id: None,
+            foreign_members: None,
+        }
+    })
+}
+
+fn get_stop_geom(stop: &Arc<gtfs_structures::Stop>) -> Option<geojson::Geometry> {
+    match (&stop.longitude, &stop.latitude) {
+        (Some(lon), Some(lat)) => Some(geojson::Geometry::new(Point(vec![*lon, *lat]))),
+        _ => None,
+    }
+}
+
+fn get_related_stop_ids(issue: &issues::Issue) -> Vec<String> {
+    let related_objects = &issue.related_objects;
+    related_objects
+        .iter()
+        .filter(|o| o.object_type == Some(ObjectType::Stop))
+        .map(|s| s.id.clone())
+        .collect()
+}
+
+fn geojson_feature_line_string(
+    stop1_id: &String,
+    stop2_id: &String,
+    gtfs: &Gtfs,
+    issue: &issues::Issue,
+) -> Option<Feature> {
+    let stop1 = gtfs.stops.get(stop1_id);
+    let stop2 = gtfs.stops.get(stop2_id);
+
+    match (stop1, stop2) {
+        (Some(stop1), Some(stop2)) => {
+            let geom = line_geometry_between_stops(stop1, stop2);
+            let mut properties = Map::new();
+
+            if issue.details.is_some() {
+                properties.insert(
+                    String::from("details"),
+                    to_value(issue.details.as_ref().clone()).unwrap(),
+                );
+            }
+
+            Some(Feature {
+                geometry: geom,
+                bbox: None,
+                properties: Some(properties),
+                id: None,
+                foreign_members: None,
+            })
+        }
+        _ => None,
+    }
+}
+
+fn line_geometry_between_stops(
+    stop1: &Arc<gtfs_structures::Stop>,
+    stop2: &Arc<gtfs_structures::Stop>,
+) -> Option<geojson::Geometry> {
+    match (
```

Comment:
You can convert a `Vec<Option>` into a `Option<Vec>`, like https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4b66ea10af25334c4f36a0097dc46186

Then you only need to map have your geometry. No need for any match

Reply:
Thanks for the advice, I didn't manage to write it that way though...and I think I prefere the more explicit aspect of the current code.